tests/scale/README.md

```diff
@@ -0,0 +1,154 @@
+# Scale testing
+## Overview
+This folder contains scale tests and related/explicit scale helpers to test OSM.
+
+Scale testing will have several use cases:
+- It will seek to qualify and document the behavior of OSM (and related control plane services) under certain loads associated to one or more dimensions (number of services, number of policies, rate of change, etc).
+- It will help find, understand and validate soft ceilings and bottlenecks on current design and implementation.
+- It should also provide instrumentation to gate at PR level (possibly an on-demand pipeline), in order to find possible performance/scale regressions before merge.
+- It should also provide meaningful insight on how the performance and scalability of the overall product offering evolves at each milestone/release.
+
+## Design
+
+Current tests leverage the existing test framework to deploy OSM and repeat a work operation (iteration) till the validation phase of the iteration fails. How and what does the iteration do is up to the test, so implementation is free to scale any resource/s till failure.
+
+The framework provides helpers to track profiling information through the overall test and individual iterations. Some of most relevant metrics are automatically captured during the test, but more exclusive metrics for speicific tests might have to be implemented. 
+Gathering these metrics require a Prometheus instance scraping both the envoy endpoints and the K8s api servers.
+
+The current set of metrics aimed to be automatically tracked across iteration for any set of resources specified in the test are:
```

Comment:
Is anything recorded about the cluster characteristics, like total CPUs/memory/nodes available? Seeing results for the same test against a kind cluster and a beefy "real" cluster might be confusing without that context.

Reply:
we should do, didn't go the extra mile as might not be immediately needed for a nightly pipeline (i imagine we will not be changing setups that often for the next months).
I'd say that the cpu limits and ram allocated in osm's deployment file is probably important too.